backend/kernelCI_app/management/commands/process_pending_aggregations.py

```diff
@@ -182,22 +204,176 @@ def _process_build_status(
+        key = _get_tree_listing_key(
+            origin=build_checkout.origin,
+            tree_name=build_checkout.tree_name,
+            git_url=build_checkout.git_repository_url,
+            git_branch=build_checkout.git_repository_branch,
+            entity_id=build_id,
+        )
+        keys_to_check.add(key)
+        build_keys_by_id[build_id] = key
+
+    return keys_to_check, test_keys_by_id, build_keys_by_id
+
+
+def _aggregate_tree_listing(
+    ready_tests: Sequence[PendingTest],
+    test_builds_by_id: dict[str, Builds],
+    ready_builds: Sequence[PendingBuilds],
+    build_checkouts_by_id: dict[str, Checkouts],
+) -> tuple[dict[str, TreeListingRecord], set[ProcessedListingItems]]:
+    """
+    Prepares tree_listing data from pending tests and builds.
+
+    Returns a dictionary of treeListing records
+    keyed by checkout_id (for updating TreeListing table)
+    and a set of new processed entries (for updating the ProcessedListingItems table).
+    """
+    if not ready_tests and not ready_builds:
+        return {}, set()
+
+    tree_listing_data: dict[str, TreeListingRecord] = {}
+    new_processed_entries: set[ProcessedListingItems] = set()
+
+    keys_to_check, test_keys_by_id, build_keys_by_id = _collect_tree_listing_keys(
+        ready_tests,
+        test_builds_by_id,
+        ready_builds,
+        build_checkouts_by_id,
+    )
+
+    existing_processed = _get_existing_processed(keys_to_check)
+
+    for test in ready_tests:
+        test_checkout_id = test_builds_by_id[test.build_id].checkout_id
+        try:
+            status_record = tree_listing_data[test_checkout_id]
+        except KeyError:
+            test_checkout = test_builds_by_id[test.build_id].checkout
+            status_record = _init_tree_listing_record(checkout=test_checkout)
+            tree_listing_data[test_checkout_id] = status_record
+
+        _process_test_status(
+            test=test,
+            test_listing_key=test_keys_by_id[test.test_id],
+            checkout_id=test_checkout_id,
+            status_record=status_record,
+            existing_processed=existing_processed,
+            new_processed_entries=new_processed_entries,
+        )
+
+    for build in ready_builds:
+        build_checkout_id = build_checkouts_by_id[build.build_id].id
+        try:
+            status_record = tree_listing_data[build_checkout_id]
+        except KeyError:
+            build_checkout = build_checkouts_by_id[build.build_id]
+            status_record = _init_tree_listing_record(checkout=build_checkout)
+            tree_listing_data[build_checkout_id] = status_record
+
+        _process_build_status(
+            build=build,
+            build_listing_key=build_keys_by_id[build.build_id],
+            status_record=status_record,
+            existing_processed=existing_processed,
+            new_processed_entries=new_processed_entries,
+        )
+
+    return tree_listing_data, new_processed_entries
+
+
+def _collect_test_contexts(
```

Comment:
add `platform` or `hardware` to the name to avoid confusion with more general tests...

since it's called by `aggregate_hardware_status` I suggest `_collect_hardware_status_contexts()`